src/components/lint/resourcelimits.ts

```diff
@@ -0,0 +1,61 @@
+import * as vscode from 'vscode';
+
+import { LinterImpl, Syntax } from './linter.impl';
+import { warningOn, childSymbols } from './linter.utils';
+
+// Pod->spec (which can also be found as Deployment->spec.template.spec)
+// .containers[each].resources.limits.{cpu,memory}
+
+export class ResourceLimitsLinter implements LinterImpl {
+    async lint(document: vscode.TextDocument, syntax: Syntax): Promise<vscode.Diagnostic[]> {
+        const resource = syntax.load(document.getText());
+        if (!resource) {
+            return [];
+        }
+
+        const podSpecPrefix =
+            resource.kind === 'Pod' ? 'spec' :
+            resource.kind === 'Deployment' ? 'spec.template.spec' :
+            undefined;
+        if (!podSpecPrefix) {
+            return [];
+        }
+
+        const symbols = await syntax.symbolise(document);
+        const containersSymbols = symbols.filter((s) => s.name === 'containers' && s.containerName === podSpecPrefix);
+        if (!containersSymbols) {
+            return [];
+        }
+
+        const warnings: vscode.Diagnostic[] = [];
+        const warnOn = (symbol: vscode.SymbolInformation, text: string) => {
+            warnings.push(warningOn(symbol, text));
+        };
+
+        for (const containersSymbol of containersSymbols) {
+            const imagesSymbols = childSymbols(symbols, containersSymbol, 'image');
+            const resourcesSymbols = childSymbols(symbols, containersSymbol, 'resources');
+            if (resourcesSymbols.length < imagesSymbols.length) {
+                warnOn(containersSymbol, 'One or more containers do not have resource limits - this could starve other processes');
```

Comment:
Is it possible for this text to render links? Might be useful to link to k8s documentation about resource starvation.

Reply:
Doesn't look like it - the type of the message is `text` and VS Code doesn't appear to autolink `http://` substrings.